### PR TITLE
[LRA-10] Post Product Saga

### PR DESCRIPTION
**Pull request by Shalin21**
feat(Create Post Product Saga): Create post saga
refactor(Create Post Product Saga): Refactor of current form
refactor(Create Post Product Saga): Refactor of IProduct interface

### CODE REVIEW

File: src/Components/Forms/CreateProductForm/CreateProductForm.tsx [hunk 1]
```diff
@@ -17,57 +44,65 @@ export const CreateProductForm = () => (
         product_details: '',
         product_price: '',
       }}
-      validationSchema={Yup.object({})}
+      validationSchema={AddProductSchema}
       onSubmit={values => {
-        //should make dispatch here
-        alert(values);
+        let newProduct: IProduct = {
```
Comment (mkozachok) [inline src/Components/Forms/CreateProductForm/CreateProductForm.tsx]: Usually name on BE an FE should be the same. 
We should skip this mapping. Please rename product props like on BE.

File: src/Components/Forms/CreateProductForm/CreateProductForm.tsx [hunk 2]
```diff
@@ -1,14 +1,41 @@
...
 import {FormBlock} from './styled';
+import { addProductStart } from '../../../store/actions/productActions';
+import { IProductState, IProduct } from '../../../types/product';
 
 interface IInput {
   field: FieldInputProps<string>,
 }
 
-export const CreateProductForm = () => (
+const AddProductSchema = Yup.object().shape({
```
Comment (mkozachok) [inline src/Components/Forms/CreateProductForm/CreateProductForm.tsx]: Please move this to  file 'validation.ts' in this folder

**Review by mkozachok — CHANGES_REQUESTED** [review]

File: src/Components/Forms/CreateProductForm/CreateProductForm.tsx (hunk 1 above)
Comment (Shalin21) [reply src/Components/Forms/CreateProductForm/CreateProductForm.tsx]: Fixed

File: src/Components/Forms/CreateProductForm/CreateProductForm.tsx (hunk 2 above)
Comment (Shalin21) [reply src/Components/Forms/CreateProductForm/CreateProductForm.tsx]: Schema moved to validation.ts in this folder

**Pull request merged**